src/Dialect/ONNX/ShapeInference/ArgMin.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------- ArgMin.cpp - Shape Inference for ArgMin Op ----------------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file implements shape inference for the ONNX ArgMin Operator.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+#include "src/Support/Diagnostic.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+LogicalResult ONNXArgMinOpShapeHelper::computeShape(
+    ONNXArgMinOpAdaptor operandAdaptor) {
+  // Get info about input data operand.
+  Value data = operandAdaptor.data();
+  int64_t dataRank = data.getType().cast<ShapedType>().getRank();
+  int64_t axisValue = op->axis();
+
+  // axis attribute must be in the range [-r,r-1], where r = rank(data).
+  if (axisValue < -dataRank || axisValue >= dataRank)
+    return onnx_mlir::Diagnostic::attributeOutOfRange(*op->getOperation(),
```

Comment:
Should the check code be put into verifier?

Reply:
Yes, we need to diagnose `axis` out of range both during verification and during shape inference. 
In some cases the rank if the input tensor is not known during verification and so by adding the diagnostic also during shape inference we have a chance to report this problem when we know the shape of the input.